welltestpy/tools/diagnostic_plots.py

```diff
@@ -0,0 +1,83 @@
+"""
+welltestpy subpackage to make diagnostic plots.
+
+.. currentmodule:: welltestpy.tools.diagnostic_plots
+
+The following classes and functions are provided
+
+.. autosummary::
+   diagnostic_plot_pump_test
+
+"""
+# pylint: disable=C0103
+
+
+import numpy as np
+
+from ..process import processlib
+from . import plotter
+
+import matplotlib.pyplot as plt
+
+
+def diagnostic_plot_pump_test(observation, fig=None, ax=None,plotname=None,style="WTP"):
+    """plot the derivative with the original data.
+
+               Parameters
+               ----------
+               observation : :class:`welltestpy.data.Observation`
+                    The observation to calculate the derivative.
+
+
+
+                Returns
+                ---------
+                Diagnostic plot
+          """
+    derivative = processlib.smoothing_derivative(observation)
+    head, time = observation()
+    head = np.array(head, dtype=float).reshape(-1)
+    time = np.array(time, dtype=float).reshape(-1)
+
+    # setting variables
+    x = time
+    y = head
+    sx = time
+    sy = head
+    dx = time[1:-1]
+    dy = derivative[1:-1]
+
+    # plotting
+    if style == "WTP":
+        style = "ggplot"
+        font_size = plt.rcParams.get("font.size", 10.0)
+        keep_fs = True
+    with plt.style.context(style):
+        if keep_fs:
+            plt.rcParams.update({"font.size": font_size})
+        fig, ax = plotter._get_fig_ax(fig, ax)
+        ax.scatter(x, y, color="red", label="drawdown")
+        ax.plot(sx, sy, c="red")
+        ax.plot(dx, dy, c="black", linestyle='dashed')
+        ax.scatter(dx, dy, c="black", marker="+",  label="time derivative")
+        ax.set_xscale("symlog")
+        ax.set_yscale("symlog")
```

Comment:
We need to determine the bounds for the symlog to have meaningful cut-off values.

Reply:
Do you mean with linthresh values for the symlogs?

Reply:
Yes. Maybe we could use arguments with standard values: (`linthresh_time=1`, linthresh_head=`1e-5`)
```python
ax.set_xscale("symlog", linthresh=linthresh_time)
ax.set_yscale("symlog", linthresh=linthresh_head)
```
Then we could also use this as the lower bound for the axis (to omit the linear part of the (sym)log-log plot).
For the upper bound, we need to determine the order of magnitude from head and time.

But what happens if you leave the bound settings untouched? Matplotlib is quite smart with this.